modular_skyrat/modules/modular_implants/code/soulcatcher/soulcatcher_component.dm

```diff
@@ -348,3 +405,17 @@ GLOBAL_LIST_EMPTY(soulcatchers)
 		return TRUE
 
 	return ..()
+
+/// Returns a list of all soulcatcher rooms beholden to this soulcatcher that user can join. Returns an empty list if user.can_join_soulcatcher is false.
+/datum/component/soulcatcher/proc/get_open_rooms(mob/user)
+	var/list/datum/soulcatcher_room/rooms = list()
+
+	if(!user.can_join_soulcatcher(src))
+		return rooms
+
```

Comment:
you misunderstand how the _room proc works - the FALSE being put in forces it to NOT check parent

this is because it allows us to do this exact logic path, where we check the parent once, first, and dont check it repeatadly for every room
this is simply the more optimized way to do it